### ISSUE TITLE
server_debian does not correctly handle alternate data_directory settings

### ISSUE THREAD

**Issue by AltxSF** [sous-chefs/postgresql#426]
### Cookbook version
6.1.1
### Chef-client version
12.18.3
### Platform Details
ubuntu 16.04
### Scenario:
Install postgres with non-default data_directory
### Steps to Reproduce:
set data_directory to something other than the default
node.override[:postgresql][:config][:data_directory] = '/postgres/data'

I believe the problem is the following snippet from recipe/server_debian.rb

```
execute 'Set locale and Create cluster' do
  command 'export LC_ALL=C; /usr/bin/pg_createcluster --start ' + node['postgresql']['version'] + ' main'
  action :run
  not_if { ::File.directory?('/etc/postgresql/' + node['postgresql']['version'] + '/main') }
end
```

The command fails to include the --datadir argument, the end result of this is that while postgresql.conf refers to the non-standard data_directory, this directory remains empty as pg_createcluster creates the various data files in its default data directory location (/var/lib/postgresql/9.5/main).

adding "--datadir" + node['postgreql]['config']['data_directory'] to the end of the command would solve this problem.

**Comment by shortdudey123** [comment]
edit: fixed formatting of chef resource code block

**Comment by damacus** [comment]
Hey @AltxSF sorry for the long triage on this one!

Can you try the new custom resource `postgresql_server_install`? 

Right now it's not configurable, but with a little tweaking, it could be fixed to do be.

**Comment by damacus** [comment]
This should now be fixed as part of #481 and is now on master. 

Please reopen if this isn't the case!
